paasta_tools/secret_providers/__init__.py

```diff
@@ -49,4 +50,11 @@ def get_secret_signature_from_data(self, data: Mapping[str, Any]) -> Optional[st
 
 
 class SecretProvider(BaseSecretProvider):
-    pass
+    # from .vault import CryptoKey # cannnot import of cyclical dependency
```

Comment:
if we do stick with this dummy attribute, https://peps.python.org/pep-0484/#runtime-or-type-checking might help type this correctly

Reply:
we could also move the typeddict into this file since if `get_key_versions` is meant to be part of the public interface, we probably want to actually define what that should look like :)